pkg/config/types/v1alpha1/v1alpha1_validation.go

```diff
@@ -50,11 +52,52 @@ var (
 
 // NetworkDeviceCheck defines the function type for checks.
 // nolint: dupl
-type NetworkDeviceCheck func(*machine.Device) error
+type NetworkDeviceCheck func(machine.Device) error
```

Comment:
Iterating over all devices and passing in a pointer made the `scopelint` linter mad. Just make this pass by value.